### ISSUE TITLE
Support parsing further keypresses and refactor key events

### ISSUE THREAD

**Issue by melloc** [jclulow/node-ansiterm#2]
It would be nice to detect and emit:

- Holding shift and control when pressing arrow keys
- More control characters (e.g. ^L, ^P, ^W, etc.)
- Function keys (F1, F2, and so on)
- ^V followed by another control character to allow entering literal values as a normal `"keypress"` event

Currently, special keypresses get emitted as separate events. This can be difficult to work with, so I'm proposing switching to only having several events:

- `"keypress"`, for normal keys
- `"control"`, for pressing control sequences
- `"arrow"`, for pressing arrow keys
- `"meta"`, for pressing special keys on the keyboard, like `"home"`, `"end"`, etc.

**Comment by jclulow** [comment]
It might be good not to overload the name `meta` here, as it's not unreasonable to use that name to describe Meta-Key combinations -- e.g., `M-x` would result in the sequence (`ESC`, `x`) on the wire.

I'm also not sure what to do about Shift-Tab, for instance.  It seems reasonable to represent Tab as {{^I}} through the {{control}} event, but what about Shift-Tab?  I believe that turns into (`ESC`, `[`, `Z`) on the wire.  Perhaps we should _also_ emit the more symbolic `tab` via whatever the `meta` event is called, with `shift` set appropriately?

**Comment by jclulow** [comment]
I suspect it would also be worthwhile to make support for `^V` something that the consumer can enable or disable.

**Comment by melloc** [comment]
I've created a pull request that contains the changes I've made for this. I decided to remove the `^V` bits, and leave that up to applications to implement. (Many applications that work with ^V for entering literal characters like to show an indicator to make the state clear to the user, like a reverse video `^` character.)

I thought about *Shift+Tab* for a while, and wasn't really satisfied with indicating it as a tab character with a shift modifier. After thinking about it for a while, I decided to represent it as a `"special"` event, named `"reverse-tab"`, to line up with how XTerm and others refer to it.

While making these changes, I used [this documentation](http://invisible-island.net/xterm/ctlseqs/ctlseqs.html) for information on some of the keypresses, and modifier combinations.
